Packages/Tracking Preview/HandRays/Runtime/Scripts/FarFieldLayerManager.cs

```diff
@@ -0,0 +1,75 @@
+/******************************************************************************
+ * Copyright (C) Ultraleap, Inc. 2011-2022.                                   *
+ *                                                                            *
+ * Use subject to the terms of the Apache License 2.0 available at            *
+ * http://www.apache.org/licenses/LICENSE-2.0, or another agreement           *
+ * between Ultraleap and you, your company or other organization.             *
+ ******************************************************************************/
+
+using UnityEngine;
+
+namespace Leap.Unity.Preview.HandRays
+{
+    /// <summary>
+    /// Assigns the correct layer to far field objects
+    /// </summary>
+    public class FarFieldLayerManager : MonoBehaviour
+    {
+        [Tooltip("If enabled, assigns the next available layer to FarFieldObjectLayer")]
+        public bool AutomaticFarFieldObjectLayer = true;
+        [Tooltip("The layer to be assigned to all far field objects")]
+        public SingleLayer FarFieldObjectLayer;
+        [Tooltip("The layer assigned to the floor")]
+        public SingleLayer FloorLayer;
+
+        private bool _layersGenerated = false;
+
+        // Start is called before the first frame update
+        void Awake()
+        {
+            GenerateLayers();
+            AssignLayers();
+        }
+
+        protected void GenerateLayers()
+        {
+            if (_layersGenerated || !AutomaticFarFieldObjectLayer)
+            {
+                return;
+            }
+
+            FarFieldObjectLayer = -1;
+
+            for (int i = 8; i < 32; i++)
```

Comment:
Presumably we are doing 8 here to skip the first known (always used layers)

Reply:
Yeah it skips the first known Unity-set layers